packages/zero-react/src/use-query.tsx

```diff
@@ -23,19 +23,38 @@ export function useQuery<
   TReturn,
 >(
   q: Query<TSchema, TTable, TReturn>,
-  enable: boolean = true,
+  enableOrOptions:
+    | boolean
+    | {
+        enable?: boolean | undefined;
+        serverSnapshot: TReturn | undefined;
+      }
+    | undefined = true,
 ): QueryResult<TReturn> {
   const z = useZero();
+  const enable =
+    typeof enableOrOptions === 'boolean'
+      ? enableOrOptions
+      : enableOrOptions.enable ?? false;
+  const options =
+    typeof enableOrOptions === 'boolean' ? undefined : enableOrOptions;
+  const serverSnapshot = options?.serverSnapshot;
   const view = viewStore.getView(
     z.clientID,
     q as AdvancedQuery<TSchema, TTable, TReturn>,
     enable && z.server !== null,
+    serverSnapshot !== undefined,
   );
   // https://react.dev/reference/react/useSyncExternalStore
   return useSyncExternalStore(
     view.subscribeReactInternals,
     view.getSnapshot,
-    view.getSnapshot,
+    serverSnapshot
```

Comment:
Do you have an example of how someone uses zero on the server to populate this?

Something like this?

```ts
const q = z.query.issue.related('assignee');
export async function getServerSideProps() {
  return { issues: await q.run() }
}

function MyComponent({issues}) {
  const q = useQuery(q, {serverSnapshot: issues});
  ...
}
```

Unclear to me how you hydrate a tree of components with SSR without one giant query at the root.